internal/services/compute/parse/community_gallery_image_version.go

```diff
@@ -0,0 +1,125 @@
+		fmt.Sprintf("Image Name %q", id.ImageName),
+		fmt.Sprintf("Version %q", id.Version),
+	}
+	segmentsStr := strings.Join(segments, " / ")
+	return fmt.Sprintf("%s: (%s)", "Community Gallery", segmentsStr)
+}
+
+func (id CommunityGalleryImageVersionId) ID() string {
+	fmtString := "/CommunityGalleries/%s/images/%s/versions/%s"
+	return fmt.Sprintf(fmtString, id.GalleryName, id.ImageName, id.Version)
+}
+
+// CommunityGalleryImageVersionID parses a CommunityGalleryImageVersion Unique ID into an CommunityGalleryImageVersionId struct
+func CommunityGalleryImageVersionID(input string) (*CommunityGalleryImageVersionId, error) {
+	segments := make([]resourceids.Segment, 0)
+
+	segments = append(segments, resourceids.Segment{
+		FixedValue: utils.String("CommunityGalleries"),
+		Name:       "communityGalleries",
+		Type:       resourceids.StaticSegmentType,
+	})
+
+	segments = append(segments, resourceids.Segment{
+		ExampleValue: "myGalleryName",
+		Name:         "galleryName",
+		Type:         resourceids.UserSpecifiedSegmentType,
+	})
+
+	segments = append(segments, resourceids.Segment{
+		FixedValue: utils.String("images"),
+		Name:       "images",
+		Type:       resourceids.StaticSegmentType,
+	})
+
+	segments = append(segments, resourceids.Segment{
+		ExampleValue: "myImageName",
+		Name:         "imageName",
+		Type:         resourceids.UserSpecifiedSegmentType,
+	})
+
+	segments = append(segments, resourceids.Segment{
+		FixedValue: utils.String("versions"),
+		Name:       "versions",
+		Type:       resourceids.StaticSegmentType,
+	})
+
+	segments = append(segments, resourceids.Segment{
+		ExampleValue: "myImageVersion",
+		Name:         "version",
+		Type:         resourceids.UserSpecifiedSegmentType,
+	})
+
+	newParser := resourceids.NewParser(segments)
+
+	id, err := newParser.Parse(input, false)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceId := CommunityGalleryImageVersionId{
+		GalleryName: id.Parsed["galleryName"],
+		ImageName:   id.Parsed["imageName"],
+		Version:     id.Parsed["version"],
+	}
+
+	if resourceId.GalleryName == "" {
+		return nil, fmt.Errorf("ID was missing the 'GalleryName' element")
+	}
+
+	if resourceId.ImageName == "" {
+		return nil, fmt.Errorf("ID was missing the 'ImageName' element")
+	}
+
+	if resourceId.Version == "" {
+		return nil, fmt.Errorf("ID was missing the 'Version' element")
+	}
+
+	// Additional validation for version, it can be the word "latest" or
+	// a string in the format of Major.Minor.Patch, it must always be
+	// a semantic version...
+
+	if !strings.EqualFold(resourceId.Version, "latest") {
+		versionParts := strings.Split(resourceId.Version, ".")
+
+		if len(versionParts) != 3 {
+			return nil, fmt.Errorf("ID 'Version' element is invalid, 'Version' must either be 'latest' or the semantic version(Major.Minor.Patch) for the image, got %s", resourceId.Version)
+		}
+
+		for _, v := range versionParts {
+			if _, err := strconv.Atoi(v); err != nil {
```

Comment:
double-checking: can users specify `1.1.x` for the latest 1.x release or something?

Reply:
No, it has to be a versionless ID(which infers latest), the literal lowercased string "latest" or a full semantic version(major.minor.patch), partials are not supported at all by the API.